controllers/module_nmc_reconciler.go

```diff
@@ -250,14 +286,12 @@ func (mnrh *moduleNMCReconcilerHelper) enableModuleOnNode(ctx context.Context, m
 	return nil
 }
 
-func (mnrh *moduleNMCReconcilerHelper) disableModuleOnNode(ctx context.Context, modNamespace, modName, nodeName string) error {
-	nmc, err := mnrh.nmcHelper.Get(ctx, nodeName)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			// NodeModulesConfig does not exists, module was never running on the node, we are good
-			return nil
-		}
-		return fmt.Errorf("failed to get the NodeModulesConfig for node %s: %v", nodeName, err)
+func (mnrh *moduleNMCReconcilerHelper) disableModuleOnNode(ctx context.Context, modNamespace, modName, nodeName string, nmcExists bool) error {
```

Comment:
Do we need that new boolean parameter? Could we move the `if` to `Reconcile`?

Reply:
I prefer to leave it in the disableModuleOnNode function. The reconciler decides whether module should be removed or added, and the appropriate functions (enableModuleOnNode and disableModuleOnNode) decide who it must be done. It is up to disableModuleOnNode to decide how to remove the module if NMC does not exists

Reply:
This boolean is part of the input to the function that should decide how to delete module. If , for example, the decision to do nothing was based on nmcExists and another parameter, would we want to do that check in the Reconcile function? What if in the future we would like to change the decision? The decision should be done in one function and not spread through different functions, IMHO